clippy_lints/src/manual_assert.rs

```diff
@@ -50,32 +51,31 @@ impl<'tcx> LateLintPass<'tcx> for ManualAssert {
             // Should this have a config value?
             && !is_else_clause(cx.tcx, expr)
         {
-            let mut applicability = Applicability::MachineApplicable;
-            let mut comments = span_extract_comment(cx.sess().source_map(), expr.span);
-            if !comments.is_empty() {
-                comments += "\n";
-            }
-            let cond_sugg = !sugg::Sugg::hir_with_context(cx, cond, expr.span.ctxt(), "..", &mut applicability);
-            let semicolon = if is_parent_stmt(cx, expr.hir_id) { ";" } else { "" };
-            let sugg = format!("assert!({cond_sugg}, {format_args_snip}){semicolon}");
-            // we show to the user the suggestion without the comments, but when applying the fix, include the
-            // comments in the block
             span_lint_and_then(
                 cx,
                 MANUAL_ASSERT,
                 expr.span,
                 "only a `panic!` in `if`-then statement",
                 |diag| {
-                    // comments can be noisy, do not show them to the user
+                    let mut applicability = Applicability::MachineApplicable;
+                    let mut comments = span_extract_comment(cx.sess().source_map(), expr.span);
                     if !comments.is_empty() {
-                        diag.tool_only_span_suggestion(
-                            expr.span.shrink_to_lo(),
-                            "add comments back",
-                            comments,
-                            applicability,
-                        );
+                        comments += "\n";
                     }
-                    diag.span_suggestion(expr.span, "try instead", sugg, applicability);
+                    let cond_sugg = !sugg::Sugg::hir_with_context(cx, cond, expr.span.ctxt(), "..", &mut applicability);
+                    let semicolon = if is_parent_stmt(cx, expr.hir_id) { ";" } else { "" };
+
+                    let indent = indent_of(cx, expr.span);
+                    let full_sugg = reindent_multiline(
+                        format!("{comments}assert!({cond_sugg}, {format_args_snip}){semicolon}").as_str(),
+                        true,
+                        indent,
+                    );
+                    diag.multipart_suggestion(
+                        "replace `if`-then-`panic!` with `assert!`",
+                        vec![(expr.span, full_sugg)],
+                        applicability,
```

Comment:
Actually, since this is just always a single suggestion now, can't we just use `.span_suggestion()` now (or rather `.span_suggestion_verbose()` to keep having it on a separate line)? That should be identical but removes the need for creating a vec here.

```suggestion
                    diag.span_suggestion_verbose(
                        expr.span,
                        "replace `if`-then-`panic!` with `assert!`",
                        full_sugg,
                        applicability,
                    );
```